enhancements/installer/assisted-installer-in-cluster.md

```diff
@@ -0,0 +1,580 @@
+1. The Baremetal Agent Controller sets the Agent's Role field in its spec to "master" or "worker" if a corresponding label was present on its BareMetalHost.
+1. The Agent is automatically marked as Approved via a field in its Spec based on being recognized as running on the known BareMetalHost.
+1. The Agent runs through the validation and inspection phases. The results are shown on the Agent's Status, and eventually a condition marks the Agent as "ready".
+1. Cluster Creator creates a ClusterDeployment describing a new cluster. It describes how many control-plane and worker agents to expect. It also includes a label selector to match Agent resources that should be part of the cluster.
+1. Cluster Creator applies a label to Agents if necessary so that they match the ClusterDeployment's selector.
+1. Once there are enough ready Agents of each role to fulfill the expected number as expressed on the ClusterDeployment, installation begins.
+
+#### Static Networking
+
+Some customers have asked for the ability to provide static network details
+up-front for each host instead of using DHCP. They want to define this
+configuration at the same time they define the corresponding BareMetalHost.
+
+A new resource called NMStateConfig will have a Spec with the following
+fields:
+
+* MACAddress: a MAC address for any network device on the host to which this config should be applied. This value is only used to ensure that the config is applied to the intended host.
+* Config: a byte array that can contain a raw [nmstate](https://www.nmstate.io/) network config.
+
+nmstate is already in use within OpenShift for applying network configuration
+to nodes via the [kubernetes-nmstate
+operator](https://github.com/nmstate/kubernetes-nmstate). It uses a similar
+pattern of embedding a raw nmstate yaml structure as a byte array in the Spec.
+
+Each NMStateConfig resource will have a label that corresponds to a InstallEnv.
+The raw YAML configs for each matching resource will be rendered to a network
+config by the Assisted Service and then embedded into the discovery ISO for
+that InstallEnv. At runtime, the discovery ISO will find the config that
+matches a MAC address on the current host and then apply the config. It does
+not matter which interface has the matching MAC address; the matching is merely
+used to identify that the current host corresponds to a given config.
+
+The NMStateConfig resource design is being discussed [in a
+pull request](https://github.com/openshift/assisted-service/pull/969).
+
+#### Install Device
+
+The Agent resource Spec will include a field on which to specify the storage device
+where the OS should be installed. That field must be set by a platform-specific
+controller or some other actor that understands the underlying host.
```

Comment:
For every field that has a corresponding field in install-config.yaml, I wonder how we're modelling those in our Hive APIs

In this case, I wonder about how this field is analogous to the new field that bootstrap-in-place has added for the installation device

Reply:
In the InstallConfig it looks like this:

```
// BootstrapInPlace defines the configuration for bootstrap-in-place installation
type BootstrapInPlace struct {
	// InstallationDisk is the target disk drive for coreos-installer
	InstallationDisk string `json:"installationDisk"`
}
```

In our case here with the agent, it could be used on day 1 or day 2, and it's more of an API to `coreos-installer` than `openshift-install`.